go/arrow/csv/writer.go

```diff
@@ -50,11 +53,20 @@ func NewWriter(w io.Writer, schema *arrow.Schema, opts ...Option) *Writer {
 func (w *Writer) Schema() *arrow.Schema { return w.schema }
 
 // Write writes a single Record as one row to the CSV file
-func (w *Writer) Write(record array.Record) error {
+func (w *Writer) Write(record array.Record) (err error) {
```

Comment:
named return values are always a bit error prone.
I'd just keep simply `error` (not `(err error)`) and define a `var err error` before the `if w.header` line.